test/test_images/latencymako/main.go

```diff
@@ -0,0 +1,471 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/google/mako/go/quickstore"
+	"knative.dev/eventing/test/common"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	vegeta "github.com/tsenart/vegeta/lib"
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/test/mako"
+	pkgpacers "knative.dev/pkg/test/vegeta/pacers"
+)
+
+const (
+	defaultEventType   = "perf-test-event-type"
+	defaultEventSource = "perf-test-event-source"
+	warmupRps          = 100
+	defaultDuration    = 10 * time.Second
+)
+
+// flags for the image
+var (
+	sinkURL       string
+	msgSize       int
+	workers       uint64
+	sentCh        chan sentState
+	deliveredCh   chan deliveredState
+	receivedCh    chan receivedState
+	resultCh      chan eventStatus
+	paceFlag      string
+	warmupSeconds uint
+	verbose       bool
+	fatalf        func(f string, args ...interface{}) = log.Fatalf
+	printf        func(f string, args ...interface{}) = func(f string, args ...interface{}) {
+		if verbose {
+			log.Printf(f, args...)
+		}
+	}
+)
+
+// eventStatus is status of the event delivery.
+type eventStatus int
+
+const (
+	sent eventStatus = iota
+	received
+	undelivered
+	dropped
+	duplicated // TODO currently unused
+	corrupted  // TODO(Fredy-Z): corrupted status is not being used now
+)
+
+type requestInterceptor struct {
+	before func(*http.Request)
+	after  func(*http.Request, *http.Response, error)
+}
+
+func (r requestInterceptor) RoundTrip(request *http.Request) (*http.Response, error) {
+	if r.before != nil {
+		r.before(request)
+	}
+	res, err := http.DefaultTransport.RoundTrip(request)
+	if r.after != nil {
+		r.after(request, res, err)
+	}
+	return res, err
+}
+
+type state struct {
+	eventId uint64
+	failed  bool
+	at      time.Time
+}
+
+type sentState state
+type deliveredState state
+type receivedState state
+
+func init() {
+	flag.BoolVar(&verbose, "verbose", false, "Enable verbose logging")
+	flag.StringVar(&sinkURL, "sink", "", "The sink URL for the event destination.")
+	flag.IntVar(&msgSize, "msg-size", 100, "The size of each message we want to send. Generate random strings to avoid caching.")
+	flag.UintVar(&warmupSeconds, "warmup", 10, "Duration of warmup phase. During warmup latencies are not recorded. 0 means no warmup")
+	flag.StringVar(&paceFlag, "pace", "100:10", "Pace array comma separater. Format rps[:duration=1s]. Example 100:2,200:4,100:1,500:60")
+	flag.Uint64Var(&workers, "workers", 1, "Number of vegeta workers")
+}
+
+type paceSpec struct {
+	rps      int
+	duration time.Duration
+}
+
+func main() {
+	// parse the command line flags
+	flag.Parse()
+
+	if paceFlag == "" {
+		fatalf("pace not set!")
+	}
+
+	if sinkURL == "" {
+		fatalf("sink not set!")
+	}
+
+	pacerSpecs, err := parsePaceSpec()
+
```

Comment:
(nit) There is usually no new line between the function call and the error handling, but I don't know about any written rule.